### ISSUE TITLE
4.11.9 input type change causing build errors

### ISSUE THREAD

**Issue by clov3r** [ionic-team/ionic#20290]
In 4.11.8 -> 4.11.9, IonicInput was updated so that value now is `string | number | undefined | null` instead of `string | | undefined | null`.

Immediately broke all my angular templates, because CI pulls in the latest patch version.

I have a whole lot of this kinda thing going on:
```
<ion-item>
  <ion-input #anInput></ion-input>

  <ion-note *ngIf=" anInput.value && anInput.value!.length < 6 ">
    The entered value is not long enough!
  </ion-note>
</ion-item>
```

and now I have this:

<img width="1552" alt="Screen Shot 2020-01-24 at 11 58 42am" src="https://user-images.githubusercontent.com/3375161/73095977-01424b00-3ea1-11ea-980c-b4c49cb657d6.png">

Why not save this to roll into v5, or at least make it a minor version bump, so the `~` version operator would stop the breakage? Don't know if you can fix this now, but wanted to point out the semver implications for the future.

Thanks for all the work that goes into this framework though, appreciate it ;)

**Comment by liamdebeasi** [comment]
Hey there, thanks for letting us know. The change that caused this was intended to be a fix for Ionic React users only. That being said, the breaking change was definitely unintentional. We are going to revert this change for now and will push an update. I will post an update here when that has been released.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This issue has been resolved via https://github.com/ionic-team/ionic/pull/20292 and is currently available as part of Ionic Framework 4.11.10.

**Comment by clov3r** [comment]
You rock 👍🏻
